pkg/services/ngalert/state/state_tracker.go

```diff
@@ -150,16 +180,33 @@ func (st *StateTracker) cleanUp() {
 }
 
 func (st *StateTracker) trim() {
-	st.Log.Info("trimming alert state cache")
+	st.Log.Info("trimming alert state cache", "now", time.Now())
 	st.stateCache.mu.Lock()
 	defer st.stateCache.mu.Unlock()
 	for _, v := range st.stateCache.cacheMap {
 		if len(v.Results) > 100 {
 			st.Log.Debug("trimming result set", "cacheId", v.CacheId, "count", len(v.Results)-100)
-			newResults := make([]eval.State, 100)
+			newResults := make([]StateEvaluation, 100)
 			copy(newResults, v.Results[100:])
 			v.Results = newResults
 			st.set(v)
 		}
 	}
 }
+
+func (a AlertState) Equals(b AlertState) bool {
```

Comment:
Because Results are not persisted to the DB they are not currently considered when testing for equality